rmw_fastrtps_shared_cpp/test/test_rmw_qos_to_dds_attributes.cpp

```diff
@@ -94,7 +95,27 @@ TEST_F(GetDataReaderQoSTest, nominal_conversion) {
   EXPECT_EQ(
     eprosima::fastrtps::KEEP_LAST_HISTORY_QOS,
     subscriber_attributes_.topic.historyQos.kind);
-  EXPECT_EQ(10, subscriber_attributes_.topic.historyQos.depth);
+  EXPECT_GE(10, subscriber_attributes_.topic.historyQos.depth);
+}
+
+TEST_F(GetDataReaderQoSTest, large_depth_conversion) {
+  size_t depth = subscriber_attributes_.topic.historyQos.depth + 1;
+  qos_profile_.depth = depth;
+  qos_profile_.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+
+  EXPECT_TRUE(get_datareader_qos(qos_profile_, subscriber_attributes_));
+
+  EXPECT_EQ(
+    eprosima::fastrtps::KEEP_LAST_HISTORY_QOS,
+    subscriber_attributes_.topic.historyQos.kind);
+  EXPECT_GE(depth, static_cast<size_t>(subscriber_attributes_.topic.historyQos.depth));
+
+  using depth_type = decltype(subscriber_attributes_.topic.historyQos.depth);
+  size_t max_depth = static_cast<size_t>(std::numeric_limits<depth_type>::max());
```

Comment:
I think `max_depth` can be declared constexpr, which means the comparison in the if statement would be evaluated at compile time and removed if not needed. Alternatively, you could use `std::is_same<size_t, depth_type>::value` in the if for the same effect (if that has the same behavior you want).

Would it make sense to add an EXPECT_TRUE check for `qos_profile_.depth = max_depth`?

Reply:
Good call ! it also made me realize I had those `EXPECT_*` backwards. See https://github.com/ros2/rmw_fastrtps/pull/457/commits/9f24d76d5370e2f83a67cbec9aaed8da9351c2fb.